lisa/sut_orchestrator/azure/credential.py

```diff
@@ -147,12 +154,47 @@ def get_credential(self) -> Any:
         """
         return AzureCredential with related schema
         """
-        return DefaultAzureCredential(cloud=self._cloud)
+        additional_tenants = ["*"] if self._allow_all_tenants else None
+        return DefaultAzureCredential(
+            cloud=self._cloud,
+            additionally_allowed_tenants=additional_tenants,
+        )
 
     def _get_key(self) -> str:
         return f"{self._credential_type}_{self._client_id}_{self._tenant_id}"
 
 
+class AzureWorkloadIdentityCredential(AzureCredential):
+    """
+    Class to create azure WorkloadIdentityCredential
+    """
+
+    @classmethod
+    def type_name(cls) -> str:
+        return AzureCredentialType.WorkloadIdentityCredential
+
+    @classmethod
+    def type_schema(cls) -> Type[schema.TypedSchema]:
+        return AzureCredentialSchema
+
+    def __init__(
+        self,
+        runbook: AzureCredentialSchema,
+        logger: Logger,
+        cloud: Cloud = AZURE_PUBLIC_CLOUD,
+    ) -> None:
+        super().__init__(runbook, cloud=cloud, logger=logger)
```

Comment:
Because there is no extra logic in the `__init__`. the `__init__` method can be removed.